crates/red_knot_python_semantic/src/unpack.rs

```diff
@@ -47,9 +49,15 @@ pub(crate) struct Unpack<'db> {
 }
 
 impl<'db> Unpack<'db> {
-    /// Returns the scope where the unpacking is happening.
-    pub(crate) fn scope(self, db: &'db dyn Db) -> ScopeId<'db> {
-        self.file_scope(db).to_scope_id(db, self.file(db))
+    /// Returns the scope where the unpacked value is appeared.
+    /// In comprehension, the scope of the value and the target may be different.
```

Comment:
I think we should be clear in which cases it's going to be different as there's just one case - first generator in a generator expression or list / dict / set comprehension